### PR TITLE
Fix previousAddress sometimes undefined

### LINKED ISSUE #293: Fix `account_address` parameter sometimes emitted with '' in GA event
## Context

The `account_address` parameter emitted as part of the 'GA4 event | wallet_connect' is sometimes just `''` which breaks the integration in GA. The parameter was introduced in https://github.com/pendulum-chain/vortex/pull/255. 

## TODO 

Change the code so that the event is not emitted without that parameter.

### PR DESCRIPTION
**Pull request by ebma**
Closes #293.

### CODE REVIEW

File: src/contexts/events.tsx [hunk 1]
```diff
@@ -153,6 +154,14 @@ const useEvents = () => {
     trackedEventTypes.current = new Set();
   }, []);
 
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (gianfra-t) [reply src/contexts/events.tsx:157]: Well in that case, it would send a `connect` event but with no address.  
I think it would be better to have the update logic only on one `useEffect` if possible, otherwise it's hard to follow. 

And in fact it could lead to deadlock (maybe not the case here).

[the remark this answers is not in the document]
Comment (ebma) [reply src/contexts/events.tsx]: True, I missed that. I changed it again and I think it's fixed now.

**Review by TorstenStueber — APPROVED** [review]
👍
